deepchem/models/torch_models/layers.py

```diff
@@ -6150,4 +6150,186 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
+    the "support".  The AttnLSTMEmbedding is thus a type of learnable
+    metric that allows a network to modify its internal notion of
+    distance.
+    See references [1]_ [2]_ for more details.
+    References
+    ----------
+    .. [1] Vinyals, Oriol, et al. "Matching networks for one shot learning."
+        Advances in neural information processing systems. 2016.
+    .. [2] Vinyals, Oriol, Samy Bengio, and Manjunath Kudlur. "Order matters:
+        Sequence to sequence for sets." arXiv preprint arXiv:1511.06391 (2015).
+    """
+    def __init__(self, n_test, n_support, n_feat, max_depth, **kwargs):
+        """
+        Parameters
+        ----------
+        n_support: int
+            Size of support set.
+        n_test: int
+            Size of test set.
+        n_feat: int
+            Number of features per atom
+        max_depth: int
+            Number of "processing steps" used by sequence-to-sequence for sets model.
+        """
+        super(AttnLSTMEmbedding, self).__init__(**kwargs)
+        self.max_depth = max_depth
+        self.n_test = n_test
+        self.n_support = n_support
+        self.n_feat = n_feat
+
+        n_feat = self.n_feat
+        self.lstm = LSTMStep(n_feat, 2 * n_feat)
+        self.q_init = torch.zeros([self.n_test, n_feat])
+        self.states_init = self.lstm.get_initial_states([self.n_test, n_feat])
+        self.built = True
+
+    def __repr__(self):
+        return (
+            f'{self.__class__.__name__}(n_test={self.n_test}, n_support={self.n_support}, n_feat={self.n_feat}, max_depth={self.max_depth})'
+        )
+    
+    def forward(self, inputs):
+        """Execute this layer on input tensors.
+        Parameters
+        ----------
+        inputs: list
+            List of two tensors (X, Xp). X should be of shape (n_test,
+            n_feat) and Xp should be of shape (n_support, n_feat) where
+            n_test is the size of the test set, n_support that of the support
+            set, and n_feat is the number of per-atom features.
+        Returns
+        -------
+        list
+            Returns two tensors of same shape as input. Namely the output
+            shape will be [(n_test, n_feat), (n_support, n_feat)]
+        """
+        if len(inputs) != 2:
+            raise ValueError("AttnLSTMEmbedding layer must have exactly two parents")
+        
+        # x is test set, xp is support set.
+        x,xp=inputs
+        x,xp=torch.tensor(x), torch.tensor(xp)
+
+        # Get initializations
+        p = self.p_init
+        q = self.q_init
+
+        z = xp
+        states = self.support_states_init
+        x_states = self.test_states_init
+
+        for d in range(self.max_depth):
+            e=cosine_dist(z+q,xp)
+            a=torch.softmax(e,dim=-1)
+            r=torch.matmul(a, xp)
+
+            y = torch.concatenate([q, r], axis=1)
+            q, states = self.lstm([y] + states)
+        
+        return [x + q, xp]
```

Comment:
Please remove the empty lines after this, only one empty line at the end is the convention in Python.

Reply:
removed